File: distributeur-cpp/src/dab/Distributeur.cpp
#include <hpms/dab/Distributeur.hpp>

#include <os/Thread.hpp>
#include <util/Time.hpp>
#include <util/Log.hpp>

#include <stdio.h>

using namespace hpms::dab;

static void * launchUI( void * arg ) {
   DistributeurUI * ui = (DistributeurUI *)arg;
   ui->run();
   return nullptr;
}

Distributeur::Distributeur( void ) :
   _ui( *this )
{
   UTIL_LOG_HERE();
   _etatDuDab->etat = dabtypes::Etat::MAINTENANCE;
   _etatDuDab->soldeCaisse = 0.0;
   os::Thread( launchUI, &_ui );
}

void Distributeur::etatDuDabPublished( void ) {
   UTIL_LOG_HERE();
   _ui.refresh();
}

void Distributeur::ejecterLaCarte( void ) {
   UTIL_LOG_HERE();
   _ui.refresh();
}

void Distributeur::ejecterLesBillets( const double & montant ) {
	UTIL_LOG_ARGS( "montant = %7.2f", montant );
   _ui.refresh();
}

void Distributeur::confisquerLaCarte( void ) {
   UTIL_LOG_HERE();
   _ui.refresh();
}

void Distributeur::placerLesBilletsDansLaCorbeille( void ) {
   UTIL_LOG_HERE();
   _ui.refresh();
}

void Distributeur::shutdown( void ) {
   UTIL_LOG_HERE();
   _ui.refresh();
}

File: Distributeur-java-script/src/hpms/dab/script/Distributeur.java
package hpms.dab.script;

import java.io.IOException;

public class Distributeur extends hpms.dab.script.DistributeurComponent {

   public Distributeur( byte instanceID ) {
      super( instanceID );
   }

   @Override
   public void etatDuDabPublished() throws IOException {
   }

   @Override
   public void confisquerLaCarte() throws IOException {
   }

   @Override
   public void placerLesBilletsDansLaCorbeille() throws IOException {
   }

   @Override
   public void ejecterLaCarte() throws IOException {
   }

   @Override
   public void arret() throws IOException {
   }

   @Override
   public void ejecterLesBillets( double montant ) throws IOException {
   }
}
